var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -71,11 +71,13 @@ class Llvm(CMakePackage):
 
     # Build dependency
     depends_on('cmake@3.4.3:', type='build')
+    depends_on('python@2.7:2.8', when='@:4.999 ~python', type='build')
+    depends_on('python', when='@5: ~python', type='build')
 
     # Universal dependency
-    depends_on('python@2.7:2.8', when='@:4.999')
-    depends_on('python')
-    depends_on('py-lit', type=('build', 'run'))
+    depends_on('python@2.7:2.8', when='@:4.999+python')
+    depends_on('python', when='@5:+python')
+    depends_on('py-lit', type='build', when='+python')
```

Comment:
If py-lit is used for testing, should it be an "always" build dep instead of a "+python" build dep?

Reply:
If it is "always" build, `python` appears in the runtime dependencies of `llvm`.

`[blr ]              ^python@2.7.16%gcc@7.3.0+bz2+ctypes+dbm+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter~ucs4~uuid+zlib arch=linux-ubuntu18.04-x86_64 
`

Which packages should I try to build to test if this `python` dependency creates an incompatibility?

Reply:
I actually removed `py-lit` as a dependency. It is actually already in llvm ([see here](https://github.com/llvm-mirror/llvm/tree/master/utils/lit)) and it is more complicated to add it from spack then to use the one directly included in llvm. Also, remove it removes that one problem that I discussed above about the python dependency.

Reply:
Even better